### PR TITLE
Refactoring for dcrCouldRender

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

Two things:

1. Abstracts away the logic required to decide the routing to DCR or not, from article app to common. It was nicely nested in the article app, but we now need to logic for updating the `guardian.window.config(.page)` object and, unless duplicating the logic to common, it now needs to live there. The change is less complicated that what it might look like actually. The code is just moved over.

2. Changes the way `guardian.window.config.page.dcrCouldRender` is generated. No longer do we patch the `guardian.window.config` object after it is created, but we simply update some functions signatures to pass it where it needs to be known.

### CODE REVIEW

File: article/app/services/dotcomponents/ArticlePicker.scala [hunk 1]
```diff
@@ -1,91 +1,9 @@
 package services.dotcomponents
 
-import controllers.ArticlePage
+import dotcomrendering.DotcomRenderingUtils._
```
Comment (jeteve) [inline article/app/services/dotcomponents/ArticlePicker.scala]: why 'Utils'?

Comment (shtukas) [reply article/app/services/dotcomponents/ArticlePicker.scala]: No real reason. I could easily rename it. Actually let me rename it :)

Comment (shtukas) [reply article/app/services/dotcomponents/ArticlePicker.scala]: Done!

**Review by jeteve — APPROVED** [review]

**Comment by shtukas** [comment]
I have just noticed that this PR, as it stands, has not yet incorporated this change ( https://github.com/guardian/frontend/pull/22010/files ), which I did afterwards. Commit coming.

**Comment by mxdvl** [comment]
Hi there!
Should this PR remain open? It's been around for a while and should probably either ship or be closed.
If it needs to remain open, could you please change the base branch to main? See mention above for context.

🧹💨 @shtukas @jeteve

**Pull request closed without merging**